runtime/tests/checker/interface_test.go

```diff
@@ -3702,14 +3734,206 @@ func TestCheckInterfaceDefaultMethodsInheritance(t *testing.T) {
+
+		_, err := ParseAndCheck(t, `
+            struct interface Foo {
+                access(all) fun hello() {
+                    pre { true }
+                }
+            }
+
+            struct interface Bar: Foo {
+                access(all) fun hello() {
+                    pre { true }
+                }
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("condition in parent", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface Foo {
+                access(all) fun hello() {
+                    pre { true }
+                }
+            }
+
+            struct interface Bar: Foo {
+                access(all) fun hello()
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("condition in child", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface Foo {
+                access(all) fun hello()
+            }
+
+            struct interface Bar: Foo {
+                access(all) fun hello() {
+                    pre { true }
+                }
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("conditions from two paths", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                access(all) fun hello() {
+                    pre { true }
+                }
+            }
+
+            struct interface B {
+                access(all) fun hello() {
+                    pre { true }
+                }
+            }
+
+            struct interface C: A, B {}
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("conditions from two paths, concrete type", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                access(all) fun hello() {
+                    pre { true }
+                }
+            }
+
+            struct interface B {
+                access(all) fun hello() {
+                    pre { true }
+                }
+            }
+
+            struct interface C: A, B {}
+
+            struct D: C {
+                access(all) fun hello() {
+                    var a = 1
+                }
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("all three formats of function, interface type", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                access(all) fun hello()
+            }
+
+            struct interface B {
+                access(all) fun hello() {
+                    pre { true }
+                }
+            }
+
+            struct interface C {
+                access(all) fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface D: A, B, C {}
+        `)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		memberConflictError := &sema.InterfaceMemberConflictError{}
```

Comment:
Still feels strange that `A`'s `hello` would conflict with `C`'s here, considering that `A` is only declaring a type and has no implementation, but I understand why it works this way.